prog/kubernetes/provision_kubernetes_node.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+class Prog::Kubernetes::ProvisionKubernetesNode < Prog::Base
+  subject_is :kubernetes_cluster
+
+  def vm
+    @vm ||= Vm[frame.fetch("vm_id", nil)] || nil
+  end
+
+  def write_hosts_file_if_needed(ip = nil)
+    return unless Config.development?
+    return if vm.sshable.cmd("sudo cat /etc/hosts").match?(/#{kubernetes_cluster.endpoint}/)
+    ip = kubernetes_cluster.cp_vms.first.ephemeral_net4 if ip.nil?
+
+    vm.sshable.cmd("echo '#{ip} #{kubernetes_cluster.endpoint}' | sudo tee -a /etc/hosts")
+  end
+
+  label def start
+    name = "#{kubernetes_cluster.name.downcase}-control-plane-#{SecureRandom.alphanumeric(5).downcase}"
+
+    vm_st = Prog::Vm::Nexus.assemble_with_sshable(
+      "ubi",
+      kubernetes_cluster.projects.first.id,
+      # we should reiterate how we name the vm. some how correlate it to the vm's info.
+      name: name,
+      location: kubernetes_cluster.location,
+      size: "standard-2",
+      boot_image: "ubuntu-jammy",
+      private_subnet_id: kubernetes_cluster.private_subnet_id,
+      enable_ip4: true
+    )
+
+    current_frame = strand.stack.first
+    current_frame["vm_id"] = vm_st.id
+    strand.modified!(:stack)
+
+    kubernetes_cluster.add_cp_vm(vm_st.subject)
+    kubernetes_cluster.api_server_lb.add_vm(vm_st.subject)
+
+    hop_install_software
+  end
+
+  label def install_software
+    nap 5 unless vm.strand.label == "wait"
+    vm.sshable.cmd <<BASH
+  set -ueo pipefail
+  echo "net.ipv6.conf.all.forwarding=1\nnet.ipv6.conf.all.proxy_ndp=1\nnet.ipv4.conf.all.forwarding=1\nnet.ipv4.ip_forward=1" | sudo tee /etc/sysctl.d/72-clover-forward-packets.conf
+  sudo sysctl --system
+  sudo modprobe br_netfilter
```

Comment:
what's this for?

Reply:
Kubeadm required this for some time, but lately it was removed (https://github.com/kubernetes/kubernetes/pull/123464/files) it seems and pushed the requirement to CNIs. I'll try if we can get away without setting this now.

Reply:
Removed it and  things are fine